### PR TITLE
Minor fixes to testnet 78 migration

### PR DESCRIPTION
**Pull request by zbuc**
## Describe your changes

Configure migrations to run testnet 78 migration, add more logging, fix bug with validator migration

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

  > consensus breaking because it's a migration

### CODE REVIEW

**Comment by conorsch** [comment]
Nice, thanks for following up. I'll dig into migration testing for 78 today!

**Comment by conorsch** [comment]
Looks like the revised migration _applies_ fine, but the chain cannot restart after the migration is run. A common mistake: I did the exact same thing when writing the Testnet77 migration (https://github.com/penumbra-zone/penumbra/pull/4503) and had to follow up with a narrow fix in https://github.com/penumbra-zone/penumbra/pull/4508. 

Unsurprisingly I see that the competing Testnet77 migration in https://github.com/penumbra-zone/penumbra/pull/4564 does _not_ make this same mistake, hat-tip to @erwanor on that. But let's fix it here, so we can proceed with migration testing, and sort out conflicts in #4654 separately.

**Review by conorsch — CHANGES_REQUESTED** [review]
Needs to reset the halt bit: I'll append a commit and re-review after testing.

**Comment by zbuc** [comment]
Oof, good catch. I had refactored the migration code to de-duplicate all the common operations in migrations but then [reverted it](https://github.com/penumbra-zone/penumbra/pull/4567/commits/5f9b10ba8a90208d28a478f08735aa86b2be34d2) and forgot to add those steps back to the testnet 78 migration.

**Comment by conorsch** [comment]
> I had refactored the migration code to de-duplicate all the common operations in migrations but then [reverted it](https://github.com/penumbra-zone/penumbra/pull/4567/commits/5f9b10ba8a90208d28a478f08735aa86b2be34d2) and forgot to add those steps back to the testnet 78 migration.

Thanks for tackling this, even if it didn't land. I remain in favor of cleaning up the migration tooling (#4294), but don't mind writing more migrations as practice to inform what the shape of that cleanup should be.

**Review by conorsch — APPROVED** [review]

File: crates/bin/pd/src/migrate/testnet78.rs [hunk 1]
```diff
@@ -161,10 +182,12 @@ async fn truncate_validator_fields(delta: &mut StateDelta<Snapshot>) -> anyhow::
 ///   * Governance Signaling Proposals:
 ///    - `commit hash` (255 bytes)
 async fn truncate_proposal_fields(delta: &mut StateDelta<Snapshot>) -> anyhow::Result<()> {
+    tracing::info!("truncating proposal fields");
     let next_proposal_id: u64 = delta.next_proposal_id().await?;
 
     // Range each proposal and truncate the fields.
     for proposal_id in 0..next_proposal_id {
+        tracing::info!("truncating proposal: {}", proposal_id);
```
Comment (erwanor) [inline crates/bin/pd/src/migrate/testnet78.rs:190]: @zbuc doesn't this have the same issue of pulling data using stronger rules that existed when it was stored?

Comment (zbuc) [reply crates/bin/pd/src/migrate/testnet78.rs:190]: Yes, will follow up in a new PR

Comment (zbuc) [reply crates/bin/pd/src/migrate/testnet78.rs:190]: https://github.com/penumbra-zone/penumbra/pull/4606

**Pull request merged**